sidechain/primitives/src/traits/mod.rs

```diff
@@ -26,7 +26,7 @@ use sp_core::{blake2_256, Pair, Public, H256};
 use sp_runtime::traits::Member;
 use sp_std::{fmt::Debug, prelude::*};
 
-pub trait Header {
+pub trait Header: Encode + Decode {
```

Comment:
Should also have `Clone` ? Since we're returning a header reference now from the block, we should have `Clone` for the header.